### ISSUE TITLE
Wybrana metoda dostawy nie wymaga podania numeru iban

### ISSUE THREAD

**Issue by allegrofan** [allegro/allegro-api#9136]
Cześć,
coś się niedawno zmieniło bo przesyłki Allegro Kurier DPD pobranie generują komunikat:
ERROR: Wybrana metoda dostawy nie wymaga podania numeru iban, wpłatę za przesyłki pobraniowe prześlemy na Twoje środki

Czy to oznacza, że wcale nie powinienem przekazywać:

  $data[additionalServices][cashOnDelivery][value][currency]="PLN";
  $data[additionalServices][cashOnDelivery][accountNumber]="PL..........";
  $data[additionalServices][cashOnDelivery][name]="........";
  $data[additionalServices][cashOnDelivery][express]=false;    

czy wtedy etykieta wygeneruje się z COD?

**Comment by Lukasz-Zurek** [comment]
Tak. Zmienił się tutaj tylko sposób wypłaty pieniędzy dla wybranych metod dostawy, w tym również dla _Allegro Kurier DPD_. Więcej informacji znajdziesz w [tym artykule](https://help.allegro.com/sell/pl/a/zmienimy-sposob-wyplaty-pieniedzy-za-zamowienia-oplacone-przy-odbiorze-na-allegro-pl-VLz97MYYjCZ). Dla tych metod nie przekazuj numeru IBAN.

**Comment by Lukasz-Zurek** [comment]
@allegrofan Tak.

**Comment by Maczuga** [comment]
Czy można prosić o informację dla jakich typów przesyłek wymagane jest dalej podawanie nr. IBAN?

**Comment by Lukasz-Zurek** [comment]
@Maczuga Tak, wspomniane pole "forceRequireIban" wskazuje na to, czy dla danej usługi dostawy wymagany jest IBAN.

**Comment by allegrofan** [comment]
forceRequireIban nie widać w parcel-management/delivery-services rozumeim że należy przejść na https://api.{environment}/shipment-management/delivery-services ?

**Comment by Lukasz-Zurek** [comment]
Tak będzie najlepiej, ponieważ nie rozwijamy już zasobu **/parcel-management/delivery-services**.

**Comment by allegrofan** [comment]
To ciekawe, po zmianie z https://api.allegro.pl/parcel-management/delivery-services na https://api.allegro.pl/shipment-management/delivery-services {"errors":[{"code":"AccessDenied","message":"Access is denied","details":null,"path":null,"userMessage":"No access to the specified resource."}]} - w https://apps.developer.allegro.pl/ mam przypisane wszystkie uprawnienia

**Comment by jezewskiBNetrox** [comment]
> To ciekawe, po zmianie z https://api.allegro.pl/parcel-management/delivery-services na https://api.allegro.pl/shipment-management/delivery-services {"errors":[{"code":"AccessDenied","message":"Access is denied","details":null,"path":null,"userMessage":"No access to the specified resource."}]} - w https://apps.developer.allegro.pl/ mam przypisane wszystkie uprawnienia

@allegrofan Wygeneruj nowe tokeny. Token zawiera zhashowane uprawnienia w sobie, więc przypisanie nowych uprawnień do aplikacji nie wystarczy. Musi powstać nowy token, który będzie miał w sobie uprawnienia które przypisałeś.

**Comment by allegrofan** [comment]
Czy wystarczy refreshAccessToken? Jeśli nie to jak wygenerować nowe?

**Comment by Lukasz-Zurek** [comment]
@allegrofan Refresh token zawiera dotychczasowe uprawnienia (nie uwzględnia wszelkich ewentualnych zmian w ustawieniach aplikacji), więc tak jak wspomniał już **jezewskiBNetrox** - musisz ponownie przejść przez proces autoryzacji od początku. [Tutaj](https://developer.allegro.pl/tutorials/uwierzytelnianie-i-autoryzacja-zlq9e75GdIR) znajdziesz instrukcje krok po kroku, zarówno dla [authorization code](https://developer.allegro.pl/tutorials/uwierzytelnianie-i-autoryzacja-zlq9e75GdIR#autoryzacja-uzytkownika), jak i dla [device](https://developer.allegro.pl/tutorials/uwierzytelnianie-i-autoryzacja-zlq9e75GdIR#device-flow-autoryzacja-uzytkownika).

**Comment by ejsiik** [comment]
Podepnę się. Gdy przekazuję pusty string w miejsce accountNumber to wciąż otrzymuję wiadomość iż "Wybrana metoda dostawy nie wymaga podania numeru iban".
![image](https://github.com/allegro/allegro-api/assets/72543889/8d52ea9f-48be-491c-b806-210ab4abb506)
![image](https://github.com/allegro/allegro-api/assets/72543889/e4f72515-98eb-4f92-8b8d-ae32e1214282)

**Comment by Lukasz-Zurek** [comment]
@ejsiik Prześlij proszę trace-id. Testowałem wczoraj na sandboxie i taki request przeszedł bez wspomnianego błędu. Dopytam zespołu odpowiedzialnego za "Wysyłam z Allegro", bo najwyraźniej mamy pewną niespójność.

**Comment by ejsiik** [comment]
trace-id: 5bc8e2194088334f
Nie jest to odosobniony przypadek, dzieje się tak przy każdej przesyłce za pobraniem kurierem dpd. Jeśli będzie potrzebne więcej to podeślę kolejne.

**Comment by allegrofan** [comment]
a u manie działa :)
